File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/persistence/MovieDatabase.java
package com.github.rahmnathan.localmovies.app.persistence;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

@Database(entities = {com.github.rahmnathan.localmovies.app.persistence.MovieEntity.class}, version = 1)
public abstract class MovieDatabase extends RoomDatabase {
    public abstract com.github.rahmnathan.localmovies.app.persistence.MovieDAO movieDAO();

    private static MovieDatabase INSTANCE;

    public static MovieDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (MovieDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context, MovieDatabase.class, "movie_database").build();
                }
            }
        }
        return INSTANCE;
    }
}

File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/persistence/MovieDatabase.kt
package com.github.rahmnathan.localmovies.app.persistence

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [MovieEntity::class], version = 1)
abstract class MovieDatabase : RoomDatabase() {
    abstract fun movieDAO(): MovieDAO?

    companion object {
        @Volatile
        private var INSTANCE: MovieDatabase? = null
        fun getDatabase(context: Context): MovieDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                        context.applicationContext,
                        MovieDatabase::class.java,
                        "word_database"
                ).build()
                INSTANCE = instance
                instance
            }
        }
    }
}